### PR TITLE
WIP: APM-38: build process for python lambda functions

### PR DESCRIPTION
**Pull request by coreyjackwilson**
### If applied, this pull request will...
- [x] Create basic structure for lambda package based on Jordans work for multipath
- [ ] Setup with some helloworld code so have something to deploy easily

### Addresses:
* [APM-38](https://appneta.atlassian.net/browse/APM-38)
* Starts implementing the collector

### Links to any relevant articles or other resources:
* [Jordan's work on Multipath](https://github.com/appneta/pvc-ops/blob/master/containers/multipath-analysis/README.md)

### CODE REVIEW

File: bgp-consumer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM dacut/amazon-linux-python-3.6
```
Comment (JordanSlaman) [inline bgp-consumer/Dockerfile]: Might not be a great thing security-wise to use this third-party dependancy...
🤷‍♂️ 
Not sure what our alternatives would be.

File: bgp-consumer/README.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+DB_SERVERS
+DB_DOMAIN
+```
+The following environment variables are optional.
+Alerting will be disabled if the HipChat configuration is not provided, and the `VERBOSE` and `DRYRUN` environment variables default to false.
+```bash
+HC_TOKEN
+HC_ROOM
+HC_HOST
+LOSS_ALERT_BLACKLIST
```
Comment (JordanSlaman) [inline bgp-consumer/README.md]: Interesting variables here

File: bgp-consumer/README.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+`docker images`
+
+You can then run the container by specifying environment data and the container tag/image ID.
+By default it will begin the daemon in the foreground logging to stdout.
+
+```bash
+# tagged image with env file
+docker run --env-file=.env bgp-consumer
+
+# Inlined environment data (rundeck)
```
Comment (JordanSlaman) [inline bgp-consumer/README.md]: can probably remove this, or alter it.
`run -e ` lets you pass env variables to the container, which we needed when running multipath in a container but probably don't need at all for the lambdas

lambdas have other ways of inserting env data, like a json from the dashboard/API

File: bgp-consumer/bgp_consumer_daemon.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import datetime
+import time
+
+import multipath
```
Comment (JordanSlaman) [inline bgp-consumer/bgp_consumer_daemon.py]: :O

File: bgp-consumer/src/main.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        dryrun=env_data['DRYRUN']
+    )
+
+
+def every_1min(es, df_getter, alerter, verbose, dryrun):
+    pass
+
+
+# This method is invoked directly by AWS lambda at one minute intervals.
+def lambda_handler(event, context):
```
Comment (JordanSlaman) [inline bgp-consumer/src/main.py]: this function is needed, it gets called by lambda.
You can replace it's contents with a log line or something for testing

File: bgp-consumer/src/__init__.py [hunk 6]
```diff
@@ -0,0 +1 @@
+from . import main, get_dataframe, manipulations, alerts
```
Comment (JordanSlaman) [inline bgp-consumer/src/__init__.py]: delete the other relative imports besides main if they don't exist

File: bgp-consumer/README.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+python3 /bgp-consumer-dist/bgp-consumer_daemon.py
+```
+
+If it works than we can package it for AWS lambda. Lambda requires a zip with all the packages in the top level, so the zip should unpack to be all the python packages required, not a folder of all the packages required.
+```bash
+cd /bgp-consumer-dist && zip -r /host-downloads/built-bgp-consumer-lambda.zip ./*
+```
+
+
+#### EC2 quick deploy instructions
```
Comment (JordanSlaman) [inline bgp-consumer/README.md]: Probably don't need this for the lambda deploy

**Comment by coreyjackwilson** [comment]
Closed for new PR

**Pull request closed without merging**